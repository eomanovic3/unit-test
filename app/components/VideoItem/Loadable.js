/**
 *
 * Asynchronously loads the component for VideoItem
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
