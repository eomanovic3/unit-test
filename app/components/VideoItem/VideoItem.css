.video-item{
  display: flex !important;
  align-items: center !important;
  cursor: pointer;
  float: left;
  min-width: 320px;
  min-width: 320px;
  padding: 2% !important;
  border-top: 1px solid rgba(34,36,38,.15) !important;
}

.video-item.item img{
  max-width: 210px;
  min-height: 330px;
}
