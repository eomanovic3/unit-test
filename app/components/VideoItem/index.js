/**
 *
 * VideoItem
 *
 */

import React from 'react';
import './VideoItem.css';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import nomovie from '../../images/nomovie.jpg';

const TitleDiv = styled.div`
  width: 220px !important;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;

const YearDiv = styled.div`
  width: 220px !important;
  float: left;
  color: #9f9f9f;
`;

const MovieImage = styled.img`
  width: ${props =>
    !props.noMovieImageAvailableFlag ? '300px !important' : '400px !important'};
  margin: auto;
  display: block;
  float: right;
  box-shadow: 5px 5px 5px grey;
`;

class VideoItem extends React.PureComponent {
  render() {
    const { video } = this.props;
    const date = video ? video.first_air_date || video.release_date : null;
    if (
      video &&
      (video.first_air_date !== undefined ||
        video.release_date !== undefined) &&
      (video.original_title || video.original_name || video.name)
    ) {
      return (
        <Link
          to={`/detailPage/${video.first_air_date ? 'tv' : 'movie'}/${
            video.id
          }`}
          className="video-item item shadow-lg d-flex flex-column"
        >
          <MovieImage
            className="ui image"
            alt={video.overview}
            src={
              video && video.poster_path
                ? `https://image.tmdb.org/t/p/w500${video.poster_path}`
                : nomovie
            }
            noMovieImageAvailableFlag={video && video.poster_path}
          />
          {video.original_title && (!video.original_name || !video.name) ? (
            <TitleDiv>{video.original_title}</TitleDiv>
          ) : null}
          {video.original_name && (!video.original_title || !video.name) ? (
            <TitleDiv>{video.original_name}</TitleDiv>
          ) : null}
          {video.name && (!video.original_name && !video.original_title) ? (
            <TitleDiv>{video.name}</TitleDiv>
          ) : null}

          <YearDiv>
            {date ? date.toString().split('-')[0] : 'No year available'}
          </YearDiv>
        </Link>
      );
    }
    return <div />;
  }
}

VideoItem.propTypes = {
  video: PropTypes.object,
};

export default VideoItem;
