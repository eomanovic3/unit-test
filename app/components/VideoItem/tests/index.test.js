/**
 *
 * Tests for VideoItem
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { act } from 'react-dom/test-utils';

import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import VideoItem from '../index';
Enzyme.configure({ adapter: new Adapter() });

let container;
let props;
beforeEach(() => {
  container = document.createElement('Link');
  props = {
    video: {
      original_title: 'Mad Hot Ballroom',
      release_date: '2005-05-13',
      overview: 'Overview',
      id: 14358,
      poster_path: '/cVnBICXmyqlusz1iYvjvIOHxo5U.jpg',
    },
  };
  container.to = `/detailPage/movie/${props.video.id}`;
  document.body.appendChild(container);
});
afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe('<VideoItem />', () => {
  it('expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<VideoItem />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<VideoItem />);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render title', () => {
    act(() => {
      ReactDOM.render(<div>{props.video.original_title}</div>, container);
    });
  });

  it('should render image', () => {
    act(() => {
      const movieImageUrl = `https://image.tmdb.org/t/p/w500${
        props.video.poster_path
      }`;
      ReactDOM.render(
        <img
          className="ui image"
          alt={props.video.original_title}
          src={movieImageUrl}
        />,
        container,
      );
    });
  });

  it('should render date', () => {
    act(() => {
      const date = props.video
        ? props.video.first_air_date || props.video.release_date
        : null;
      const formattedDate = date ? date.toString().split('-')[0] : null;
      ReactDOM.render(<div>{formattedDate}</div>, container);
    });
  });

  it('should check props.video', () => {
    expect(container.to).toBe(`/detailPage/movie/${props.video.id}`);
  });

  it('should render with props.video', () => {
    const component = shallow(<VideoItem video={props.video} />);
    expect(component).toMatchSnapshot();
  });

  it('should render with props.video null', () => {
    props.video = null;
    const component = shallow(<VideoItem video={props.video} />);
    expect(component).toMatchSnapshot();
  });
});
