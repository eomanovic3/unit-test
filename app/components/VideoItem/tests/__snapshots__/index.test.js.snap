// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoItem /> should render and match the snapshot 1`] = `<div />`;

exports[`<VideoItem /> should render with props.video 1`] = `ShallowWrapper {}`;

exports[`<VideoItem /> should render with props.video null 1`] = `ShallowWrapper {}`;
